kaos/algorithm/view_cone.py

```diff
@@ -1,164 +1,163 @@
+      ValueError: if any of the 4 formulas has a complex answer. This happens when the orbit and
+      viewing cone do not intersect or only intersect twice.
 
-    Note:
-        With more analysis it should be possible to find a correct interval even in the case
-        where there are only two intersections but this is beyond the current scope of the project.
+    Note: With more analysis it should be possible to find a correct interval even in the case
+      where there are only two intersections but this is beyond the current scope of the project.
     """
-
-    # Get geocentric angles from site ECI
-    r_site_magnitude, lat_geoc, lon_geoc = cart2sp(site_eci.x, site_eci.y, site_eci.z)
+    lat_geoc = (lat_geoc * mp.pi) / 180
+    lon_geoc = (lon_geoc * mp.pi) / 180
 
     # P vector (also referred  to as orbital angular momentum in the paper) calculations
-    p_unit_x, p_unit_y, p_unit_z = cross(sat_pos, sat_vel) / (norm(sat_pos) * norm(sat_vel))
-
-    # Formulas from paper:
-    # Note: each Txxx represents an intersection between viewing cone and the orbit
-    gamma = THETA_NAUGHT + asin((r_site_magnitude * sin((pi / 2) + THETA_NAUGHT)) / q_magnitude)
-    tin = ((1 / ANGULAR_VELOCITY_EARTH) * (asin((cos(gamma) - (p_unit_z * sin(lat_geoc))) /
-            (sqrt((p_unit_x ** 2) + (p_unit_y ** 2)) * cos(lat_geoc)))
-            - lon_geoc - atan(p_unit_x / p_unit_y) + 2 * pi * m))
-    tout = ((1 / ANGULAR_VELOCITY_EARTH) * (pi - asin((cos(gamma) - (p_unit_z * sin(lat_geoc))) /
-            (sqrt((p_unit_x ** 2) + (p_unit_y ** 2)) * cos(lat_geoc)))
-            - lon_geoc - atan(p_unit_x / p_unit_y) + 2 * pi * m))
-
-    # Second set
-    gamma2 = pi - gamma
-    tin_2 = ((1 / ANGULAR_VELOCITY_EARTH) * (asin((cos(gamma2) - (p_unit_z * sin(lat_geoc))) /
-            (sqrt((p_unit_x ** 2) + (p_unit_y ** 2)) * cos(lat_geoc)))
-            - lon_geoc - atan(p_unit_x / p_unit_y) + 2 * pi * m))
-    tout_2 = ((1 / ANGULAR_VELOCITY_EARTH) * (pi - asin((cos(gamma2) - (p_unit_z * sin(lat_geoc))) /
-            (sqrt((p_unit_x ** 2) + (p_unit_y ** 2)) * cos(lat_geoc)))
-            - lon_geoc - atan(p_unit_x / p_unit_y) + 2 * pi * m))
-
-    return tin, tout, tin_2, tout_2
+    p_unit_x, p_unit_y, p_unit_z = cross(sat_pos, sat_vel) / (mp.norm(sat_pos) * mp.norm(sat_vel))
+
+    # Formulas from paper
+    r_site_magnitude = earth_radius_at_geocetric_lat(lat_geoc)
+    gamma1 = THETA_NAUGHT + mp.asin((r_site_magnitude * mp.sin((mp.pi / 2) + THETA_NAUGHT)) / q_max)
+    gamma2 = mp.pi - gamma1
+
+    arctan_term = mp.atan2(p_unit_x, p_unit_y)
+    arcsin_term_gamma, arcsin_term_gamma2 = [(mp.asin((mp.cos(gamma) - p_unit_z * mp.sin(lat_geoc))
+                                             / (mp.sqrt((p_unit_x ** 2) + (p_unit_y ** 2)) *
+                                              mp.cos(lat_geoc)))) for gamma in [gamma1, gamma2]]
+
+    angle_1 = (arcsin_term_gamma - lon_geoc - arctan_term + 2 * mp.pi * m)
```

Comment:
A comment explaining the angles would be appropriated or rather why we needed this ( The fact that we had to use atan2 instead of atan_